src/eos-metrics-instrumentation.c

```diff
@@ -284,6 +300,70 @@ record_live_boot (gpointer unused)
     return G_SOURCE_REMOVE;
 }
 
+static gchar *
+get_image_version_for_path (const gchar *path)
+{
+    ssize_t xattr_size = 0;
+    gchar *image_version = NULL;
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
+
+    if (xattr_size == -1)
+      return NULL;
+
+    image_version = g_malloc0 (xattr_size + 1);
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR,
+	                       image_version, xattr_size);
+
+    /* this check is just in case the xattr has changed in between the
+     * size checks */
+    if (xattr_size == -1)
+      {
+        g_warning ("Error when getting 'eos-image-version' from %s", path);
+        return NULL;
```

Comment:
I used g_autofree originally but it was freeing the return value when the function returned, as far as I could tell...